test/general/key.js

```diff
@@ -1916,6 +1916,57 @@ vqBGKJzmO5q3cECw
 =X9kJ
 -----END PGP PRIVATE KEY BLOCK-----`;
 
+
+const rsaSignOnly = `-----BEGIN PGP PRIVATE KEY BLOCK-----
+Comment: different
```

Comment:
What is the meaning of these comments? :D

Reply:
They were automatically generated by openpgpjs, not sure why... I just encrypted the message and it ended up like this, I have seen it before when using keys that were somewhat "off", but haven't looked into it too much.

Reply:
I see. I guess it's coming from this: https://github.com/openpgpjs/openpgpjs/blob/cc1bdcbae8d9f7d392d9da3fbc5ded9c42f62302/test/general/openpgp.js#L741
Maybe this is an argument for why we should undo configs in tests :upside_down_face: 
Anyway, please just remove the comments, they're a bit confusing

Reply:
> Maybe this is an argument for why we should undo configs in tests upside_down_face

Another idea would be to avoid global config object altogether and provide it explicitly during initialization (`const openpgp = require('openpgp')(config);`). That'd of course be quite a big change but worth mentioning as each test would have a clean slate by design :)

Reply:
Yeah. Even more than that, sometimes you want to set a config option for one message/decryption only, which is a bit difficult right now with the async model. So I think it would be even nicer to add `config` options/parameters to all of the top-level functions. Then you could also do:
```js
try {
  result = openpgp.decrypt({ message });
} catch (e) {
  warn("Insecure!");
  result = openpgp.decrypt({ message, config: { allow_insecure_decryption_with_signing_keys: true } });
}
```
Or make it depend on the date of the email, et cetera.
And then there would also be no statefulness in the tests, potentially allowing us to run them in parallel, etc.

Reply:
Great idea! This reminds me of the [policy system](https://docs.sequoia-pgp.org/sequoia_openpgp/policy/index.html) we use at Sequoia. Running test in parallel would be a nice side-effect too...

For top level functions this could be hacked together with a tiny try-finally around the actual code:

```js
function decrypt(..., config) {
  const oldConfig = openpgp.config;
  openpgp.config = Object.assign({}, openpgp.config, config);
  try {
    // actual code...
  } finally {
    openpgp.config = oldConfig;
  }
}
```

Reply:
> For top level functions this could be hacked together with a tiny try-finally around the actual code:

This wouldn't work if you're running multiple encryptions/decryptions in parallel, since another call might overwrite the config before it's finished. So we'd probably have to pass these config parameters throughout the codebase, unfortunately.

Reply:
Good point. Thanks for taking the time to discuss internals @twiss!